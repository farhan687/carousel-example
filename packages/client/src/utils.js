export const getNextIndex = (activeIndex, total) => (
  total > activeIndex + 1 ? activeIndex + 1 : 0
);

export const getPrevIndex = (activeIndex, total) => (
  activeIndex > 0 ? activeIndex - 1 : total - 1
);

export const groupDataInBatch = (data, size) => (
  data.reduce((batches, item) => {
    const previousBatch = batches[batches.length - 1];
    const hasSpace = previousBatch.length < size;
    if (hasSpace) {
      previousBatch.push(item);
      return batches;
    }
    return [
      ...batches,
      [item],
    ];
  }, [[]])
);

export default getNextIndex;
