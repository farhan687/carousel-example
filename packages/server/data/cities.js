const CITIES = [
  {
    "_id": "61f0b76b73b724e38be19bd3",
    "index": 0,
    "title": "Snyderville Palau",
    "images": [
      "https://picsum.photos/id/1/200/300",
      "https://picsum.photos/id/2/200/300",
      "https://picsum.photos/id/3/200/300"
    ]
  },
  {
    "_id": "61f0b76bc401b91bb8ac3b5a",
    "index": 1,
    "title": "Durham New Hampshire",
    "images": [
      "https://picsum.photos/id/4/200/300",
      "https://picsum.photos/id/5/200/300"
    ]
  },
  {
    "_id": "61f0b76b3278b9e2d8909799",
    "index": 2,
    "title": "Rosburg Pennsylvania",
    "images": [
      "https://picsum.photos/id/6/200/300",
      "https://picsum.photos/id/7/200/300"
    ]
  },
  {
    "_id": "61f0b76be1fcfa00e0f0c8ee",
    "index": 3,
    "title": "Vicksburg Delaware",
    "images": [
      "https://picsum.photos/id/8/200/300",
      "https://picsum.photos/id/9/200/300",
      "https://picsum.photos/id/10/200/300"
    ]
  },
  {
    "_id": "61f0b76b57426554fa8a86e9",
    "index": 4,
    "title": "Keller Florida",
    "images": [
      "https://picsum.photos/id/11/200/300",
      "https://picsum.photos/id/12/200/300"
    ]
  },
  {
    "_id": "61f0b76b27067fea054eeec0",
    "index": 5,
    "title": "Kidder West Virginia",
    "images": [
      "https://picsum.photos/id/13/200/300",
      "https://picsum.photos/id/14/200/300",
      "https://picsum.photos/id/15/200/300"
    ]
  },
  {
    "_id": "61f0b76b81e65365914a1a87",
    "index": 6,
    "title": "Ironton Maine",
    "images": [
      "https://picsum.photos/id/16/200/300",
      "https://picsum.photos/id/17/200/300"
    ]
  },
  {
    "_id": "61f0b76be68a00132837eb67",
    "index": 7,
    "title": "Brethren Utah",
    "images": [
      "https://picsum.photos/id/18/200/300",
      "https://picsum.photos/id/19/200/300",
      "https://picsum.photos/id/20/200/300"
    ]
  },
  {
    "_id": "61f0b76bdc65b1539cf71867",
    "index": 8,
    "title": "Tilden Wisconsin",
    "images": [
      "https://picsum.photos/id/21/200/300",
      "https://picsum.photos/id/22/200/300"
    ]
  },
  {
    "_id": "61f0b76bc5d1945281f2a19b",
    "index": 9,
    "title": "Navarre Oregon",
    "images": [
      "https://picsum.photos/id/23/200/300",
      "https://picsum.photos/id/24/200/300"
    ]
  },
  {
    "_id": "61f0b76be35ac8d6fa5a6d1e",
    "index": 10,
    "title": "Dodge California",
    "images": [
      "https://picsum.photos/id/25/200/300",
      "https://picsum.photos/id/26/200/300",
      "https://picsum.photos/id/27/200/300"
    ]
  },
  {
    "_id": "61f0b76bd457e7b297c11510",
    "index": 11,
    "title": "Unionville Louisiana",
    "images": [
      "https://picsum.photos/id/28/200/300",
      "https://picsum.photos/id/1/200/300"
    ]
  },
  {
    "_id": "61f0b76bc7591dd2cdbcbf8a",
    "index": 12,
    "title": "Alderpoint Oklahoma",
    "images": [
      "https://picsum.photos/id/29/200/300",
      "https://picsum.photos/id/30/200/300",
      "https://picsum.photos/id/31/200/300"
    ]
  },
  {
    "_id": "61f0b76b840396b481faa474",
    "index": 13,
    "title": "Montura Washington",
    "images": [
      "https://picsum.photos/id/32/200/300",
      "https://picsum.photos/id/33/200/300"
    ]
  },
  {
    "_id": "61f0b76b2d2bc3929c951c95",
    "index": 14,
    "title": "Nogal Arizona",
    "images": [
      "https://picsum.photos/id/34/200/300",
      "https://picsum.photos/id/35/200/300"
    ]
  },
  {
    "_id": "61f0b76b6548d9b614001d10",
    "index": 15,
    "title": "Carlos New York",
    "images": [
      "https://picsum.photos/id/36/200/300",
      "https://picsum.photos/id/37/200/300",
      "https://picsum.photos/id/38/200/300"
    ]
  },
  {
    "_id": "61f0b76b87fb1546b558fb5e",
    "index": 16,
    "title": "Dragoon Georgia",
    "images": [
      "https://picsum.photos/id/39/200/300",
      "https://picsum.photos/id/40/200/300"
    ]
  },
  {
    "_id": "61f0b76b8d9f1081ea88a757",
    "index": 17,
    "title": "Colton Puerto Rico",
    "images": [
      "https://picsum.photos/id/41/200/300",
      "https://picsum.photos/id/42/200/300",
      "https://picsum.photos/id/43/200/300"
    ]
  },
  {
    "_id": "61f0b76b5626fd04454368e2",
    "index": 18,
    "title": "Jessie Iowa",
    "images": [
      "https://picsum.photos/id/44/200/300",
      "https://picsum.photos/id/45/200/300",
      "https://picsum.photos/id/46/200/300"
    ]
  }
];

module.exports = CITIES;
