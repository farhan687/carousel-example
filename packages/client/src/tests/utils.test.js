import * as utils from '../utils';
import data from './__fixtures__/items.json';

describe('utils', () => {
  it('groupDataInBatch', () => {
    expect(utils.groupDataInBatch(data, 2)).toMatchSnapshot();
  });

  it('getNextIndex', () => {
    expect(utils.getNextIndex(0, 3)).toBe(1);
    expect(utils.getNextIndex(2, 3)).toBe(0);
    expect(utils.getNextIndex(4, 3)).toBe(0);
  });

  it('getPrevIndex', () => {
    expect(utils.getPrevIndex(0, 3)).toBe(2);
    expect(utils.getPrevIndex(2, 3)).toBe(1);
    expect(utils.getPrevIndex(-1, 3)).toBe(2);
  });
});
