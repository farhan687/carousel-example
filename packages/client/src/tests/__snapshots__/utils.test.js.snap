// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils groupDataInBatch 1`] = `
Array [
  Array [
    Object {
      "_id": "61f0b76b73b724e38be19bd3",
      "images": Array [
        "https://picsum.photos/id/1/200/300",
        "https://picsum.photos/id/2/200/300",
        "https://picsum.photos/id/3/200/300",
      ],
      "index": 0,
      "title": "Snyderville Palau",
    },
    Object {
      "_id": "61f0b76bc401b91bb8ac3b5a",
      "images": Array [
        "https://picsum.photos/id/4/200/300",
        "https://picsum.photos/id/5/200/300",
      ],
      "index": 1,
      "title": "Durham New Hampshire",
    },
  ],
]
`;
